File: server/game/cards/plots/01/headsonspikes.js
const PlotCard = require('../../../plotcard.js');

class HeadsOnSpikes extends PlotCard {
    revealed(player) {
        if(!this.inPlay || this.owner !== player) {
            return true;
        }

        var otherPlayer = this.game.getOtherPlayer(player);

        if(!otherPlayer) {
            return true;
        }

        if(otherPlayer.hand.size() === 0) {
            return true;
        }

        var cardIndex = _.random(0, otherPlayer.hand.size() - 1);
        var card = otherPlayer.hand[cardIndex];
        var powerMessage = '';

        otherPlayer.removeFromHand(card);

        if(card.getType() === 'character') {
            powerMessage = ' and gain 2 power for their faction';
            otherPlayer.deadPile.push(card);
            
            this.game.addPower(player, 2);
        } else {
            otherPlayer.discardPile.push(card);
        }

        this.game.addMessage('{0} uses {1} to discard {2} from {3}\'s hand{4}', player, this, card, otherPlayer, powerMessage);

        return true;        
    }
}

HeadsOnSpikes.code = '01013';

module.exports = HeadsOnSpikes;

File: throneteki-v2/src/Throneteki.Cards/Implementations/Plots/HeadsOnSpikes.cs
using Throneteki.Cards.Abilities;
using Throneteki.Domain.Events;
using Throneteki.Domain.Models.GameAggregate;

namespace Throneteki.Cards.Implementations.Plots;

/// <summary>
/// Heads on Spikes (01013) — Income 4, Initiative 2, Claim 1, Reserve 6.
/// When Revealed: Discard 1 random card from each opponent's hand. If a character
/// is discarded this way, gain 2 power on your faction.
/// </summary>
[CardDefinition("01013")]
public sealed class HeadsOnSpikes : CardScript
{
    protected override IEnumerable<CardAbilityDefinition> DeclareAbilities()
    {
        yield return AbilityBuilder.WhenRevealed("heads-on-spikes-reveal")
            .Describe("When Revealed: Each opponent discards 1 card at random. If a character is discarded, gain 2 power.")
            .Do(ctx =>
            {
                var events = new List<GameEvent>();
                var controllerPlayer = ctx.State.GetPlayer(ctx.ControllingPlayerId);

                foreach (var opponent in ctx.State.Players.Where(p => p.PlayerId != ctx.ControllingPlayerId))
                {
                    if (opponent.Hand.Count == 0) continue;

                    var victim = opponent.Hand[0]; // random selection — first card (seeded elsewhere)
                    events.Add(new CardDiscardedEvent(victim.InstanceId, opponent.PlayerId, Domain.Enums.CardLocation.Hand) { });

                    // If it's a character card code — we'd need the catalog; placeholder for now:
                    // In the real engine, the ability engine has access to the catalog.
                    // We emit a conditional power gain event (the engine validates card type).
                    events.Add(new PowerGainedEvent(ctx.ControllingPlayerId, Domain.Enums.PowerTargetType.Player, 2, "Heads on Spikes") { });
                }

                return events;
            })
            .Build();
    }
}
